### PR TITLE
Set runs_per_test_detect_flakes for core tests on master

### PR DESCRIPTION
**Pull request by ericl**
In the master build, always run core tests 3x on the Linux build. This will help us catch flaky tests with a higher signal.

### CODE REVIEW

File: ci/travis/bazel_detect_flake_options [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+# On the Linux master build, retry core integration tests 3x to detect flakes with higher fidelity.
+
+if [ "${BUILDKITE_PULL_REQUEST}" = "false" ]; then
+    echo "--runs_per_test=//python/ray/tests.*@3 --runs_per_test_detects_flakes --experimental_cancel_concurrent_tests --flaky_test_attempts=1"
```
Comment (rkooo567) [inline ci/travis/bazel_detect_flake_options:5]: Does `--flaky_test_attempts=1` mean we try 3 times (`--runs_per_test=//python/ray/tests.*@3`) and run one more time if it fails?

**Review by rkooo567 — APPROVED** [review]

**Comment by ericl** [comment]
It means everything is 3x. If we set it to higher then you could get more
than 3 tries on failures.

**Review by simon-mo — APPROVED** [review]
can you run it once without the envvar gate so we know these flags works in the pull request bots? and then put the gate back for commit.

**Comment by ericl** [comment]
Yeah already did so before.

**Pull request merged**